### DISCUSSION TITLE
Feature Request: [next-auth/react] Add ability to customize underlying

### DISCUSSION THREAD

**Discussion by william-will-angi** [nextauthjs/next-auth#12203] [Ideas]
### Goals

1. Adding better client-side support for next-auth instances that are accessed cross-origin.

### Non-Goals

1. This feature should not affect the ways that the underlying behave at all.

### Background

We are exploring a setup where we have next-auth hosted on `next-auth.app.com` as a dedicated service, while a consuming app lives on `frontend.app.com`. The problem we are running into is that the `fetch` calls will not include the next-auth cookies necessary without adding a `{ credentials: 'include' }` as an option to the request, so none of the client-side utilities work out of the box. 

There are few places where this is discussed like here:
https://github.com/nextauthjs/next-auth/discussions/1299#discussioncomment-3788577

### Proposal

It seems like the simplest solution would be to add a prop to the `SessionProvider` component, that would allow you to customize this behavior, most likely updating the underlying fetch options.

I can probably put together a PR with what I am imagining.

I am willing to contribute to a discussion of this feature as well as creating the PR if the changes are not overly complex.

**Comment by willwill96** [comment]
I created a PR with the way I was thinking about the problem: 
https://github.com/nextauthjs/next-auth/pull/12204
